Makefile

```diff
@@ -61,6 +61,17 @@ proto-format:
 build-docker:
 	$(DOCKER) build -t celestiaorg/celestia-app -f docker/Dockerfile .
 
+fmt:
+	@go mod tidy -compat=1.17
```

Comment:
should we update this to 1.18?

Reply:
🤔 it's not immediately clear based on docs

> By default, go mod tidy will check that the [selected versions](https://go.dev/ref/mod#glos-selected-version) of modules do not change when the module graph is loaded by the Go version immediately preceding the version indicated in the go directive. The versioned checked for compatibility can also be specified explicitly via the -compat flag.

from https://go.dev/ref/mod#go-mod-tidy

Given we have other Makefile tasks that already use `-compat=1.18` I think we should be consistent here. 

On second thought, we can also just remove this line entirely and rely on `make build` and `make mod` to run `go mod tidy` so will remove.